ios/RNPurchases.m

```diff
@@ -246,6 +246,69 @@ static void logUnavailablePresentCodeRedemptionSheet() {
     NSLog(@"[Purchases] Warning: tried to present codeRedemptionSheet, but it's only available on iOS 14.0 or greater.");
 }
 
+#pragma mark - Win-Back getOfferings
+RCT_EXPORT_METHOD(eligibleWinBackOffersForProductIdentifier:(nonnull NSString *)productID
+                  resolve:(RCTPromiseResolveBlock)resolve
+                  reject:(RCTPromiseRejectBlock)reject) {
+    if (@available(iOS 18.0, macOS 15.0, tvOS 18.0, watchOS 11.0, visionOS 2.0, *)) {
+        [RCCommonFunctionality eligibleWinBackOffersForProductIdentifier:productID
+                                                         completionBlock:^(NSArray<NSDictionary *> * _Nullable offers, RCErrorContainer * _Nullable errorContainer) {
+            if (errorContainer) {
+                reject(
+                    [NSString stringWithFormat:@"%ld", (long)errorContainer.code],
+                    errorContainer.message,
+                    errorContainer.error
+                );
+            } else {
+                resolve(offers ?: @[]);
+            }
+        }];
+    } else {
+        NSString *description = @"iOS win-back offers are only available on iOS 18.0 or greater.";
+        NSError *error = [[NSError alloc] initWithDomain:@"RCPurchasesErrorCodeDomain"
+                                                    code:RCUnsupportedError
+                                                userInfo:@{NSLocalizedDescriptionKey: description}];
+        reject([NSString stringWithFormat:@"%ld", (long)error.code], [error localizedDescription], error);
+    }
+}
+
+RCT_EXPORT_METHOD(purchaseProductWithWinBackOffer:(nonnull NSString *)productID
```

Comment:
Objc conventions make me think this `WithWinBackOffer` expect the next parameter to be the winback offer. I think it might be clearer without that part, and it more closely follows the method signature in PHC.

Reply:
I tried changing the name of this function to `purchaseProduct`, but then I started seeing namespace collisions with the other purchaseProduct function:

```
RNPurchases.purchaseProduct was called with 6 arguments but expects 2 arguments. If you haven't changed this method yourself, this usually means that your versions of the native code and JavaScript code are out of sync. Updating both should make this error go away.
```